### DISCUSSION TITLE
vcpkg-tool should use shortened paths for internal source path info

### DISCUSSION THREAD

**Discussion by horenmar** [microsoft/vcpkg#20581] [Ideas]
In #20515 I got an error message mentioning path `D:\a\_work\1\s\src\vcpkg\base\json.cpp(338)`, which is path relevant to the build server that compiled the vcpkg binary, but is needlessly noisy. I think that the internal source path info should instead be shortened to just `vcpkg\base\json.cpp`. 

On Linux platforms this could be easily done through `-ffile-prefix-map`, but AFAIK MSVC does not have an equivalent functionality, so vcpkg would need custom `__FILE__` replacement that cuts off the prefix.

**Comment by sylveon** [comment]
There is actually, it's /d1trimfile
